deepchem/models/torch_models/antibody_modeling.py

```diff
@@ -0,0 +1,275 @@
+    BERT based models:
+
+    [1] ProtBERT
+    [2] IgBERT
+
+    Attributes
+    ----------
+    task: str
+        The task the HuggingFaceModel is performing. Default: 'mlm'.
+    model_path: str
+        The huggingface model path of the pLM.
+    n_tasks: int
+        Number of tasks for a given model. Default: 1
+
+    Methods
+    -------
+    __init__(task, model_path, n_tasks)
+        Initialize an DeepAbLLM with specified information.
+    _mask_seq_pos(position, idx)
+        Mask a sequence at a particular index.
+    redesign_residue(n=1.0)
+        Change the photo's gamma exposure.
+    optimize_residue(n=1.0)
+        Change the photo's gamma exposure.
+    optimize_sequence(n=1.0)
+        Change the photo's gamma exposure.
+
+    Notes
+    -----
+    Model currently supports:
+    1. Light or Heavy Chain Re-Design at Arbitrary Point
+
+    (WIP)
+    2. Agnostic to Light or Heavy Chain (Depending on if the specified
+                                         model correctly accounts for this)
+
+    (Planned)
+    3. Model consumes both epitope and receptor information to influence logits
+    4. Conditional Generation (Auto-Regressive/Iterative Unmasking)
+
+    References
+    ----------
+    .. [1] Elnaggar, Ahmed, et al. "Prottrans: Toward understanding the language
+           of life through self-supervised learning." IEEE transactions on pattern
+           analysis and machine intelligence 44.10 (2021): 7112-7127.
+    .. [2] Kenlay, H., Dreyer, F. A., Kovaltsuk, A., Miketa, D., Pires, D.,
+           & Deane, C. M. (2024). Large scale paired antibody language models.
+           arXiv [q-Bio.BM]. Retrieved from http://arxiv.org/abs/2403.17889
+
+    Example Usage:
+    --------------
+    >>> # Optimize Sequence
+    >>> from deepchem.models.torch_models.antibody_modeling import DeepAbLLM
+    >>> model_path = 'Rostlab/prot_bert'
+    >>> anti_llm = DeepAbLLM(task='mlm', model_path=model_path, n_tasks=1)
+    >>> anti_llm.model.to('cuda')  # Move to GPU for faster inference
+    >>> optimized_sequences = anti_llm.optimize_sequence('GSELTQDPAVSVALGQTVRITCQGDSLRNYYASWYQQKPRQAPVLVFYGKNNRPSGIPDRFSGSSSGNTASLTISGAQAEDEADYYCNSRDSSSNHLVFGGGTKLTVLSQ', rounds=1)
+    >>> optimized_sequences
+
+    >>>>    [(0,
+            'Q',
+            'QSETQDPAVSVALGQTVRITCQGDSLRNYYASWYQQKPRQAPVLVFYGKNNRPSGIPDRFSGSSSGNTASLTISGAQAEDEADYYCNSRDSSSNHLVFGGGTKLTVLSQ',
+            0.7314766049385071) ...
+            ]
+    """
+
+    def __init__(self,
+                 task: str = "mlm",
+                 model_path: str = 'Rostlab/prot_bert',
+                 n_tasks: int = 1,
+                 **kwargs) -> None:
+        """
+        Parameters
+        ----------
+        task: str
+            The task defines the type of learning task in the model. The supported tasks are
+            - `mlm` - masked language modeling commonly used in pretraining
+            - `classification` - use it for classification tasks
+        model_path: str
```

Comment:
Change to hf_model_link/hf_model_path

Reply:
On second read: model_path accepts local or HF paths as inputs so I think this ones okay as is.